softwareComponents/configuration/src/rofibot.cpp

```diff
@@ -48,6 +55,54 @@ roficom::Orientation roficom::stringToOrientation( const std::string& str ) {
         assert( false && "String does not represent the orientation" );
 }
 
+int Component::getIndexInParent() const
+{
+    assert( parent != nullptr );
+    return parent->componentIdx( *this );
+}
+
+Matrix Component::getPosition() const
+{
+    assert( parent != nullptr );
+    assert( parent->parent != nullptr );
+    auto moduleAbsPosition = parent->parent->getModulePosition( parent->getId() );
+    return moduleAbsPosition * parent->getComponentRelativePosition( getIndexInParent() );
+}
+
+std::optional< std::pair< const Component &, roficom::Orientation > > Component::getNearConnector() const
+{
+    assert( type == ComponentType::Roficom );
+    assert( parent != nullptr );
+    assert( parent->parent != nullptr );
+
+    Rofibot & rofibot = *parent->parent;
+    if ( !rofibot.isPrepared() )
+        throw std::runtime_error( "rofibot is not prepared" );
+
+    auto thisAbsPosition = getPosition();
+    for ( auto & moduleInfo : rofibot.modules() ) {
+        assert( moduleInfo.module );
+
+        for ( const Component & nearConnector : moduleInfo.module->connectors() ) {
+            assert( nearConnector.type == ComponentType::Roficom );
```

Comment:
I have an issue with inconsistency of styles (e.g., opening brackets on a newline instead of the same line, and `T &` instead of `T&`). Otherwise it's fine by me.

Reply:
Addressed in [4eb8bdd...1ac00c2](https://github.com/paradise-fi/RoFI/compare/4eb8bddc3e832cfe0224124baa72ddffc874d582..1ac00c2ee26269312e7b13afced9d5cfe4f22cda)